File: nice_primes/nice_primes.rs
// [dependencies]
// primal = "0.3"

fn digital_root(n: u64) -> u64 {
    if n == 0 {
        0
    } else {
        1 + (n - 1) % 9
    }
}

fn nice_primes(from: usize, to: usize) {
    primal::Primes::all()
        .skip_while(|x| x < &from)
        .take_while(|x| x < &to)
        .filter(|x| primal::is_prime(digital_root(*x as u64)))
        .for_each(|x| println!("{}", x));
}

fn main() {
    nice_primes(500, 1000);
}

File: nice_primes/nice_primes.cpp
#include <iostream>

bool is_prime(unsigned int n) {
    if (n < 2)
        return false;
    if (n % 2 == 0)
        return n == 2;
    if (n % 3 == 0)
        return n == 3;
    for (unsigned int p = 5; p * p <= n; p += 4) {
        if (n % p == 0)
            return false;
        p += 2;
        if (n % p == 0)
            return false;
    }
    return true;
}

unsigned int digital_root(unsigned int n) {
    return n == 0 ? 0 : 1 + (n - 1) % 9;
}

int main() {
    std::cout << "Nice primes between 500 and 1000:\n";
    for (unsigned int n = 501; n < 1000; n += 2) {
        if (is_prime(digital_root(n)) && is_prime(n))
            std::cout << n << '\n';
    }
}
